packages/envd/internal/api/upload.go

```diff
@@ -77,33 +44,62 @@ func processFile(r *http.Request, path string, part *multipart.Part, user *user.
 
 	if err == nil {
 		if stat.IsDir() {
-			errMsg := fmt.Errorf("path is a directory: %s", path)
+			err := fmt.Errorf("path is a directory: %s", path)
 
-			return http.StatusBadRequest, errMsg
+			return http.StatusBadRequest, err
 		}
 	}
 
-	file, err := os.Create(path)
+	hasBeenChowned := false
+	err = os.Chown(path, uid, gid)
```

Comment:
@djeebus Should this chown be needed only if the stat above succeeded?

Reply:
Ah, good point, I'll add that.